packages/stealth-txs/scripts/watcher/yearn.ts

```diff
@@ -35,125 +49,218 @@ const generateRandomNumber = (min: number, max: number): string => {
+      currentGas.maxPriorityFeePerGas > MAX_PRIORITY_FEE_GAS_PRICE
+        ? utils.parseUnits(`${MAX_PRIORITY_FEE_GAS_PRICE}`, 'gwei')
+        : utils.parseUnits(`${currentGas.maxPriorityFeePerGas}`, 'gwei'),
   };
-  const populatedTx = await stealthVault.populateTransaction.reportHash(hash, { ...gasParams, gasLimit: 1000000, nonce });
-  const signedtx = await web3.eth.accounts.signTransaction(
+  console.log('[Reporting hash] Max fee per gas:', utils.formatUnits(gasParams.maxFeePerGas, 'gwei'), 'gwei');
+  console.log('[Reporting hash] Max priority fee per gas:', utils.formatUnits(gasParams.maxPriorityFeePerGas, 'gwei'), 'gwei');
+  const populatedTx = await stealthVault.populateTransaction.reportHash(hash, {
+    ...gasParams,
+    gasLimit: BigNumber.from('100000'),
+    nonce,
+  });
+  const signedTx = await web3ReporterSigner.signTransaction({
+    ...gasParams,
+    to: populatedTx.to!,
+    gas: populatedTx.gasLimit!.toNumber(),
+    data: populatedTx.data!,
+  });
+  const executorTx = getRawTransaction(transaction);
+  const bundle: FlashbotBundle = [
     {
-      ...gasParams,
-      to: stealthVaultAddress,
-      gas: '100000',
-      data: populatedTx.data!,
+      signedTransaction: signedTx.rawTransaction!,
     },
-    reporterPrivateKey as string
-  );
-  const reportingTx = await ethers.provider.sendTransaction(signedtx.rawTransaction!);
-  console.log('sent report with tx hash', reportingTx.hash);
+    {
+      signedTransaction: executorTx,
+    },
+  ];
+  if (await submitBundle(bundle)) {
+    nonce = nonce.add(1);
+  }
+}
+
+async function submitBundle(bundle: FlashbotBundle): Promise<boolean> {
+  let submitted = false;
+  let rejected = false;
+  const blockNumber = await ethers.provider.getBlockNumber();
+  let targetBlock = blockNumber + 1;
+  while (!(submitted || rejected)) {
+    alive.stillAlive();
+    if (!(await simulateBundle(bundle, targetBlock))) {
+      rejected = true;
+      continue;
+    }
+    const flashbotsTransactionResponse: FlashbotsTransaction = await flashbotsProvider.sendBundle(bundle, targetBlock);
+    flashbotsProvider.sendBundle(bundle, targetBlock + 1);
+    flashbotsProvider.sendBundle(bundle, targetBlock + 2);
+    const resolution = await (flashbotsTransactionResponse as FlashbotsTransactionResponse).wait();
+    if (resolution == FlashbotsBundleResolution.BundleIncluded) {
+      console.log('[Flashbot] BundleIncluded, sucess!');
+      submitted = true;
+    } else if (resolution == FlashbotsBundleResolution.BlockPassedWithoutInclusion) {
+      console.log('[Flashbot] BlockPassedWithoutInclusion, re-build and re-send bundle');
+    } else if (resolution == FlashbotsBundleResolution.AccountNonceTooHigh) {
+      console.log('[Flashbot] AccountNonceTooHigh, adjust nonce');
+      rejected = true;
+    }
+    targetBlock += 1;
+  }
+  return submitted;
```

Comment:
I really don't like this. But this is just a script, not an app. We could take sometime to scale this into a dapp that takes care of every case / side-case. Handles nonces and bundles in a really smart way, but thats not even close to an MVP.